proc_dash/app.py

```diff
@@ -179,5 +248,58 @@ def reset_table(contents, filename):
     raise PreventUpdate
 
 
+@app.callback(
+    [
+        Output("fig-pipeline-status-all-ses", "figure"),
+        Output("fig-pipeline-status-all-ses", "style"),
+    ],
+    Input("upload-data", "contents"),
+    State("upload-data", "filename"),
+    prevent_initial_call=True,
+)
+def generate_overview_status_fig_for_participants(contents, filename):
+    """
+    If new dataset uploaded, generate stacked bar plot of pipeline_complete statuses per session,
+    grouped by pipeline. Provides overview of the number of participants with each status in a given session,
+    per processing pipeline.
+    """
+    if contents is None:
+        raise PreventUpdate
+    data, total_subjects, sessions, upload_error = util.parse_csv_contents(
```

Comment:
You have the same section in `update_outputs`. I think a better design would be to have a single callback that fires in response to the new data upload event and as its output has a https://dash.plotly.com/dash-core-components/store element. Then you can just handle the store element as input in your figure making callbacks.

You could then also handle any csv parsing exceptions in this callback and display something helpful to the user (as I commented below)